makeStage2.sh

```diff
@@ -39,5 +40,101 @@ cd $usepop/pop/x/src/
 # ways to construct the Poplog system.
 # $usepop/pop/src/newpop -link -nox -norsv
 # $usepop/pop/src/newpop -link -x=-xm -norsv
+# $usepop/pop/src/newpop -link -x=-xt -norsv
+
+mkdir -p "${BUILD_HOME}/environments"
+
+# We need to capture the Poplog environment for each build variant. This is
+# required for the poplog command tool, which needs to bind the appropriate 
+# environment variables. Each build variant has slightly different variables
+# and values. So we capture these and later on will synthesise them into 
+# C-code functions. This is the best point at which to capture the variables.
+
+# Note that we do not wish to capture the values of variables automatically
+# introduced by running a shell SHLVL and PWD. Nor do we want to capture
+# the folder location variables poplib, poplocalauto, poplocalbin. That is
+# because GetPoplog has different defaults for those and the old values are
+# irrelevant.
+
+# We run the popenv.sh script inside a clean environment to capture the 
+# set of environment variables needed. We then need to replace any matches
+# of the string "_build/poplog_base" ($usepop) with our unique value USEPOP.
+# This will allow us to dynamically substitute with the selfHome'd value
+# at run-time. 
+#
+# This is inherently a weak strategy because it relies on being able to 
+# identify substitutions of $usepop. We improve its robustness by doing the
+# process twice with different values of $usepop - using the '..' trick.
+# If the resultant code is not identical we have a problem and we halt.
+# (N.B It is probably not necessary to run the variables through sort but I 
+# couldn't find a clear guarantee that env generates a sorted list.)
+#
+echo_env() {
+    cmd='(usepop="'"$1"'" && . $usepop/pop/com/popenv.sh && env)'
+    env -i sh -c "$cmd" | sort | \
+    grep -v '^\(_\|SHLVL\|PWD\|poplib\|poplocal\(auto\|bin\)\?\)=' | \
+    sed -e 's!'"$1"'![//USEPOP//]!g' | \
+    sort
```

Comment:
`env | sort` breaks on multi-line variable definitions. `env` does not produce a sorted list of environment variables.

Reply:
https://stackoverflow.com/questions/60756020/print-environment-variables-sorted-by-name-including-variables-with-newlines has fixes

Reply:
Yeah I will tweak the fix you sent - it won't work without a teeny bit of adjustment. Basically the newline can't go back in immediately as it causes trouble in the `sed` transformation. But it should all work out.

Reply:
By the way, you've got sort in there twice, I don't think that's necessary? The sorting should be preserved through the pipeline.

Reply:
I have removed the double-sort (well spotted!)

Reply:
At this point I am going to accept the treatment of newlines and backslashes as tech-debt on the basis that:
 - We will sort it out when we transform makePoplogCommander.sh into a Python script and we can see that coming.
- There are no plausible circumstances under which any Poplog environment variable contains a newline.
- It's not cheap to solve and not an active issue.

Reply:
I have raised Issue #97 to park this issue for a while.